.ci/scripts/go-plus/terraform-validator-tester/test_terraform_validator.sh

```diff
@@ -44,6 +44,9 @@ curl \
 
 set +e
 
+go mod edit -replace github.com/hashicorp/terraform-provider-google-beta=github.com/$github_username/terraform-provider-google-beta@$new_branch
+go mod tidy
+
```

Comment:
I'm curious what will happen now if there is a build failure here?

Will we still send the message for terraform-validator build and unit test failure? Could you test this scenario? Just want to make sure we aren't inadvertently breaking this part of the pipeline.

The UX we expect is `terraform-validator-build-and-unit-test` (or similar) should be send if there is a build failure for validator.